uncategorized SQLException; SQL state [99999]; error code [17009]; 关闭的语句; nested exception is java.sql.SQLException: 关闭的语句

### Ⅰ. Issue Description
使用若依框架微服务版集成seta1.4.2 
druid+com.baomidou:dynamic-datasource-spring-boot-starter:3.5.0多数据源
业务数据库是oracle
在AT模式下第一次访问正常 第二次访问就会报错 错误日志如下
```
org.springframework.jdbc.UncategorizedSQLException: 
### Error updating database.  Cause: java.sql.SQLException: 关闭的语句
### The error may exist in file [D:\workspace\jsyg-back\zzgl-mps\target\classes\mapper\seata\SeataProductMapper.xml]
### The error may involve com.zzgl.mps.testseata.mapper.SeataProductMapper.updateById-Inline
### The error occurred while setting parameters
### SQL: update seata_product         set price            = ?,             stock            = ?,             last_update_time = sysdate         where id = ?
### Cause: java.sql.SQLException: 关闭的语句
; uncategorized SQLException; SQL state [99999]; error code [17009]; 关闭的语句; nested exception is java.sql.SQLException: 关闭的语句
	at org.mybatis.spring.MyBatisExceptionTranslator.translateExceptionIfPossible(MyBatisExceptionTranslator.java:92)
	at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:441)
	at com.sun.proxy.$Proxy174.update(Unknown Source)
	at org.mybatis.spring.SqlSessionTemplate.update(SqlSessionTemplate.java:288)
	at com.baomidou.mybatisplus.core.override.MybatisMapperMethod.execute(MybatisMapperMethod.java:64)
	at com.baomidou.mybatisplus.core.override.MybatisMapperProxy$PlainMethodInvoker.invoke(MybatisMapperProxy.java:148)
	at com.baomidou.mybatisplus.core.override.MybatisMapperProxy.invoke(MybatisMapperProxy.java:89)
	at com.sun.proxy.$Proxy511.updateById(Unknown Source)
	at com.zzgl.mps.testseata.service.impl.SeataProductServiceImpl.reduceStock(SeataProductServiceImpl.java:46)
	at com.zzgl.mps.testseata.service.impl.SeataProductServiceImpl$$FastClassBySpringCGLIB$$6692b00b.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
	at com.zzgl.mps.testseata.service.impl.SeataProductServiceImpl$$EnhancerBySpringCGLIB$$d784d922.reduceStock(<generated>)
	at com.zzgl.mps.testseata.service.impl.SeataOrderServiceImpl.placeOrder(SeataOrderServiceImpl.java:62)
	at com.zzgl.mps.testseata.service.impl.SeataOrderServiceImpl$$FastClassBySpringCGLIB$$47d2d18c.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at io.seata.spring.annotation.GlobalTransactionalInterceptor$2.execute(GlobalTransactionalInterceptor.java:184)
	at io.seata.tm.api.TransactionalTemplate.execute(TransactionalTemplate.java:127)
	at io.seata.spring.annotation.GlobalTransactionalInterceptor.handleGlobalTransaction(GlobalTransactionalInterceptor.java:181)
	at io.seata.spring.annotation.GlobalTransactionalInterceptor.invoke(GlobalTransactionalInterceptor.java:150)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
	at com.zzgl.mps.testseata.service.impl.SeataOrderServiceImpl$$EnhancerBySpringCGLIB$$3141b645.placeOrder(<generated>)
	at com.zzgl.mps.testseata.controller.SeataOrderController.placeOrder(SeataOrderController.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:895)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1070)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:963)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:681)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:96)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:360)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:890)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1789)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.sql.SQLException: 关闭的语句
	at oracle.jdbc.driver.OracleClosedStatement.exitImplicitCacheToActive(OracleClosedStatement.java:2955)
	at oracle.jdbc.driver.OraclePreparedStatementWrapper.exitImplicitCacheToActive(OraclePreparedStatementWrapper.java:1247)
	at com.alibaba.druid.util.OracleUtils.exitImplicitCacheToActive(OracleUtils.java:94)
	at com.alibaba.druid.pool.PreparedStatementPool.get(PreparedStatementPool.java:66)
	at com.alibaba.druid.pool.DruidPooledConnection.prepareStatement(DruidPooledConnection.java:372)
	at io.seata.rm.datasource.exec.BaseTransactionalExecutor.buildTableRecords(BaseTransactionalExecutor.java:353)
	at io.seata.rm.datasource.exec.UpdateExecutor.beforeImage(UpdateExecutor.java:72)
	at io.seata.rm.datasource.exec.AbstractDMLBaseExecutor.executeAutoCommitFalse(AbstractDMLBaseExecutor.java:99)
	at io.seata.rm.datasource.exec.AbstractDMLBaseExecutor.doExecute(AbstractDMLBaseExecutor.java:84)
	at io.seata.rm.datasource.exec.BaseTransactionalExecutor.execute(BaseTransactionalExecutor.java:113)
	at io.seata.rm.datasource.exec.ExecuteTemplate.execute(ExecuteTemplate.java:111)
	at io.seata.rm.datasource.exec.ExecuteTemplate.execute(ExecuteTemplate.java:50)
	at io.seata.rm.datasource.PreparedStatementProxy.execute(PreparedStatementProxy.java:55)
	at sun.reflect.GeneratedMethodAccessor235.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.ibatis.logging.jdbc.PreparedStatementLogger.invoke(PreparedStatementLogger.java:59)
	at com.sun.proxy.$Proxy662.execute(Unknown Source)
	at org.apache.ibatis.executor.statement.PreparedStatementHandler.update(PreparedStatementHandler.java:47)
	at org.apache.ibatis.executor.statement.RoutingStatementHandler.update(RoutingStatementHandler.java:74)
	at org.apache.ibatis.executor.SimpleExecutor.doUpdate(SimpleExecutor.java:50)
	at org.apache.ibatis.executor.BaseExecutor.update(BaseExecutor.java:117)
	at org.apache.ibatis.executor.CachingExecutor.update(CachingExecutor.java:76)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.ibatis.plugin.Plugin.invoke(Plugin.java:64)
	at com.sun.proxy.$Proxy660.update(Unknown Source)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.update(DefaultSqlSession.java:194)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:427)
	... 93 common frames omitted
18:47:33.303 [http-nio-9601-exec-4] WARN  o.s.w.s.m.m.a.ExceptionHandlerExceptionResolver - [logException,208] - Resolved [org.springframework.jdbc.UncategorizedSQLException: <EOL><EOL>### Error updating database.  Cause: java.sql.SQLException: 关闭的语句<EOL><EOL>### The error may exist in file [D:\workspace\jsyg-back\zzgl-mps\target\classes\mapper\seata\SeataProductMapper.xml]<EOL><EOL>### The error may involve com.zzgl.mps.testseata.mapper.SeataProductMapper.updateById-Inline<EOL><EOL>### The error occurred while setting parameters<EOL><EOL>### SQL: update seata_product         set price            = ?,             stock            = ?,             last_update_time = sysdate         where id = ?<EOL><EOL>### Cause: java.sql.SQLException: 关闭的语句<EOL>; uncategorized SQLException; SQL state [99999]; error code [17009]; 关闭的语句; nested exception is java.sql.SQLException: 关闭的语句]
```
网上搜索了好多方案没找到解决方法

### Ⅱ. Environment:
- JDK version(e.g. `java -version`):1.8
- Seata client/server version:  1.4.2
- Database version: Oracle
- OS(e.g. `uname -a`):
- Others:

--- a/changes/en-us/2.x.md
+++ b/changes/en-us/2.x.md
@@ -16,6 +16,7 @@ Add changes here for all PR submitted to the 2.x branch.
 - [[#6182](https://github.com/apache/incubator-seata/pull/6182)] fix guava-32.0.0-jre.jar zip file is empty in ci
 - [[#6196](https://github.com/apache/incubator-seata/pull/6196)] fix asf config file format error
 - [[#6204](https://github.com/apache/incubator-seata/pull/6204)] fix the problem that The incorrect configuration needs to be fixed
+- [[#6248](https://github.com/apache/incubator-seata/pull/6248)] fix JDBC resultSet, statement, connection closing order
 
 ### optimize:
 - [[#6031](https://github.com/apache/incubator-seata/pull/6031)] add a check for the existence of the undolog table

--- a/changes/zh-cn/2.x.md
+++ b/changes/zh-cn/2.x.md
@@ -16,6 +16,7 @@
 - [[#6182](https://github.com/apache/incubator-seata/pull/6182)] 修复在ci中guava-32.0.0-jre.jar zip文件为空的问题
 - [[#6196](https://github.com/apache/incubator-seata/pull/6196)] 修复asf配置格式错误的问题
 - [[#6204](https://github.com/apache/incubator-seata/pull/6204)] 修复错误配置问题
+- [[#6248](https://github.com/apache/incubator-seata/pull/6248)] 修复JDBC resultSet, statement, connection关闭顺序
 
 ### optimize:
 - [[#6031](https://github.com/apache/incubator-seata/pull/6031)] 添加undo_log表的存在性校验

--- a/rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java
+++ b/rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java
@@ -486,8 +486,10 @@ public abstract class BaseTransactionalExecutor<T, S extends Statement> implemen
      * @throws SQLException the sql exception
      */
     protected TableRecords buildTableRecords(TableMeta tableMeta, String selectSQL, ArrayList<List<Object>> paramAppenderList) throws SQLException {
+        PreparedStatement ps = null;
         ResultSet rs = null;
-        try (PreparedStatement ps = statementProxy.getConnection().prepareStatement(selectSQL)) {
+        try {
+            ps = statementProxy.getConnection().prepareStatement(selectSQL);
             if (CollectionUtils.isNotEmpty(paramAppenderList)) {
                 for (int i = 0, ts = paramAppenderList.size(); i < ts; i++) {
                     List<Object> paramAppender = paramAppenderList.get(i);
@@ -499,7 +501,7 @@ public abstract class BaseTransactionalExecutor<T, S extends Statement> implemen
             rs = ps.executeQuery();
             return TableRecords.buildRecords(tableMeta, rs);
         } finally {
-            IOUtil.close(rs);
+            IOUtil.close(rs, ps);
         }
     }
 
@@ -524,9 +526,10 @@ public abstract class BaseTransactionalExecutor<T, S extends Statement> implemen
         List<String> needColumns =
             getNeedColumns(tableMeta.getTableName(), sqlRecognizer.getTableAlias(), insertColumnsUnEscape);
         needColumns.forEach(selectSQLJoin::add);
+        PreparedStatement ps = null;
         ResultSet rs = null;
-        try (PreparedStatement ps = statementProxy.getConnection().prepareStatement(selectSQLJoin.toString())) {
-
+        try {
+            ps = statementProxy.getConnection().prepareStatement(selectSQLJoin.toString());
             int paramIndex = 1;
             for (int r = 0; r < rowSize; r++) {
                 for (int c = 0; c < pkColumnNameList.size(); c++) {
@@ -539,7 +542,7 @@ public abstract class BaseTransactionalExecutor<T, S extends Statement> implemen
             rs = ps.executeQuery();
             return TableRecords.buildRecords(getTableMeta(), rs);
         } finally {
-            IOUtil.close(rs);
+            IOUtil.close(rs, ps);
         }
     }
 

--- a/rm-datasource/src/main/java/io/seata/rm/datasource/exec/MultiUpdateExecutor.java
+++ b/rm-datasource/src/main/java/io/seata/rm/datasource/exec/MultiUpdateExecutor.java
@@ -127,13 +127,15 @@ public class MultiUpdateExecutor<T, S extends Statement> extends AbstractDMLBase
         }
         TableMeta tmeta = getTableMeta(sqlRecognizers.get(0).getTableName());
         String selectSQL = buildAfterImageSQL(tmeta, beforeImage);
+        PreparedStatement pst = null;
         ResultSet rs = null;
-        try (PreparedStatement pst = statementProxy.getConnection().prepareStatement(selectSQL)) {
+        try {
+            pst = statementProxy.getConnection().prepareStatement(selectSQL);
             SqlGenerateUtils.setParamForPk(beforeImage.pkRows(), getTableMeta().getPrimaryKeyOnlyName(), pst);
             rs = pst.executeQuery();
             return TableRecords.buildRecords(tmeta, rs);
         } finally {
-            IOUtil.close(rs);
+            IOUtil.close(rs, pst);
         }
     }
 

--- a/rm-datasource/src/main/java/io/seata/rm/datasource/exec/UpdateExecutor.java
+++ b/rm-datasource/src/main/java/io/seata/rm/datasource/exec/UpdateExecutor.java
@@ -100,13 +100,15 @@ public class UpdateExecutor<T, S extends Statement> extends AbstractDMLBaseExecu
             return TableRecords.empty(getTableMeta());
         }
         String selectSQL = buildAfterImageSQL(tmeta, beforeImage);
+        PreparedStatement pst = null;
         ResultSet rs = null;
-        try (PreparedStatement pst = statementProxy.getConnection().prepareStatement(selectSQL)) {
+        try {
+            pst = statementProxy.getConnection().prepareStatement(selectSQL);
             SqlGenerateUtils.setParamForPk(beforeImage.pkRows(), getTableMeta().getPrimaryKeyOnlyName(), pst);
             rs = pst.executeQuery();
             return TableRecords.buildRecords(tmeta, rs);
         } finally {
-            IOUtil.close(rs);
+            IOUtil.close(rs, pst);
         }
     }
 

--- a/rm-datasource/src/main/java/io/seata/rm/datasource/exec/mysql/MySQLInsertOnDuplicateUpdateExecutor.java
+++ b/rm-datasource/src/main/java/io/seata/rm/datasource/exec/mysql/MySQLInsertOnDuplicateUpdateExecutor.java
@@ -271,9 +271,11 @@ public class MySQLInsertOnDuplicateUpdateExecutor extends MySQLInsertExecutor im
         if (CollectionUtils.isEmpty(paramAppenderList)) {
             throw new NotSupportYetException("the SQL statement has no primary key or unique index value, it will not hit any row data.recommend to convert to a normal insert statement");
         }
+        PreparedStatement ps = null;
         ResultSet rs = null;
-        try (PreparedStatement ps = statementProxy.getConnection()
-            .prepareStatement(primaryKeys.isEmpty() ? selectSQL + " FOR UPDATE" : selectSQL)) {
+        try {
+            ps = statementProxy.getConnection()
+                    .prepareStatement(primaryKeys.isEmpty() ? selectSQL + " FOR UPDATE" : selectSQL);
             int paramAppenderCount = 0;
             int ts = CollectionUtils.isEmpty(paramAppenderList) ? 0 : paramAppenderList.size();
             for (int i = 0; i < ts; i++) {
@@ -291,7 +293,7 @@ public class MySQLInsertOnDuplicateUpdateExecutor extends MySQLInsertExecutor im
             rs = ps.executeQuery();
             return TableRecords.buildRecords(tableMeta, rs);
         } finally {
-            IOUtil.close(rs);
+            IOUtil.close(rs, ps);
         }
     }
 

--- a/rm-datasource/src/main/java/io/seata/rm/datasource/exec/mysql/MySQLUpdateJoinExecutor.java
+++ b/rm-datasource/src/main/java/io/seata/rm/datasource/exec/mysql/MySQLUpdateJoinExecutor.java
@@ -154,14 +154,16 @@ public class MySQLUpdateJoinExecutor<T, S extends Statement> extends UpdateExecu
                 continue;
             }
             String selectSQL = buildAfterImageSQL(joinTable, tableItems[i], tableBeforeImage);
+            PreparedStatement pst = null;
             ResultSet rs = null;
-            try (PreparedStatement pst = statementProxy.getConnection().prepareStatement(selectSQL)) {
+            try {
+                pst = statementProxy.getConnection().prepareStatement(selectSQL);
                 setAfterImageSQLPlaceHolderParams(joinConditionParams,tableBeforeImage.pkRows(), getTableMeta(tableItems[i]).getPrimaryKeyOnlyName(), pst);
                 rs = pst.executeQuery();
                 TableRecords afterImage = TableRecords.buildRecords(getTableMeta(tableItems[i]), rs);
                 afterImagesMap.put(tableItems[i], afterImage);
             } finally {
-                IOUtil.close(rs);
+                IOUtil.close(rs, pst);
             }
         }
         return null;

--- a/server/src/main/java/io/seata/server/console/impl/db/BranchSessionDBServiceImpl.java
+++ b/server/src/main/java/io/seata/server/console/impl/db/BranchSessionDBServiceImpl.java
@@ -81,10 +81,13 @@ public class BranchSessionDBServiceImpl implements BranchSessionService {
         String branchSessionSQL = LogStoreSqlsFactory.getLogStoreSqls(dbType).getAllBranchSessionSQL(branchTable, whereCondition);
 
         List<BranchSessionVO> list = new ArrayList<>();
-        ResultSet rs = null;
 
-        try (Connection conn = dataSource.getConnection();
-             PreparedStatement ps = conn.prepareStatement(branchSessionSQL)) {
+        Connection conn = null;
+        PreparedStatement ps = null;
+        ResultSet rs = null;
+        try {
+            conn = dataSource.getConnection();
+            ps = conn.prepareStatement(branchSessionSQL);
             ps.setObject(1, xid);
             rs = ps.executeQuery();
             while (rs.next()) {
@@ -93,7 +96,7 @@ public class BranchSessionDBServiceImpl implements BranchSessionService {
         } catch (SQLException e) {
             throw new StoreException(e);
         } finally {
-            IOUtil.close(rs);
+            IOUtil.close(rs, ps, conn);
         }
         return PageResult.success(list, list.size(), 0, 0, 0);
     }

--- a/server/src/main/java/io/seata/server/console/impl/db/GlobalLockDBServiceImpl.java
+++ b/server/src/main/java/io/seata/server/console/impl/db/GlobalLockDBServiceImpl.java
@@ -88,12 +88,15 @@ public class GlobalLockDBServiceImpl implements GlobalLockService {
         List<GlobalLockVO> list = new ArrayList<>();
         int count = 0;
 
+        Connection conn = null;
+        PreparedStatement ps = null;
+        PreparedStatement countPs = null;
         ResultSet rs = null;
         ResultSet countRs = null;
-
-        try (Connection conn = dataSource.getConnection();
-             PreparedStatement ps = conn.prepareStatement(queryLockSql);
-             PreparedStatement countPs = conn.prepareStatement(lockCountSql)) {
+        try {
+            conn = dataSource.getConnection();
+            ps = conn.prepareStatement(queryLockSql);
+            countPs = conn.prepareStatement(lockCountSql);
             PageUtil.setObject(ps, sqlParamList);
             rs = ps.executeQuery();
             while (rs.next()) {
@@ -107,7 +110,7 @@ public class GlobalLockDBServiceImpl implements GlobalLockService {
         } catch (SQLException e) {
             throw new StoreException(e);
         } finally {
-            IOUtil.close(rs, countRs);
+            IOUtil.close(rs, countRs, ps, countPs, conn);
         }
         return PageResult.success(list, count, param.getPageNum(), param.getPageSize());
     }

--- a/server/src/main/java/io/seata/server/console/impl/db/GlobalSessionDBServiceImpl.java
+++ b/server/src/main/java/io/seata/server/console/impl/db/GlobalSessionDBServiceImpl.java
@@ -96,12 +96,15 @@ public class GlobalSessionDBServiceImpl implements GlobalSessionService {
         int count = 0;
 
 
+        Connection conn = null;
+        PreparedStatement ps = null;
+        PreparedStatement countPs = null;
         ResultSet rs = null;
         ResultSet countRs = null;
-
-        try (Connection conn = dataSource.getConnection();
-             PreparedStatement ps = conn.prepareStatement(querySessionSql);
-             PreparedStatement countPs = conn.prepareStatement(sessionCountSql)) {
+        try {
+            conn = dataSource.getConnection();
+            ps = conn.prepareStatement(querySessionSql);
+            countPs = conn.prepareStatement(sessionCountSql);
             PageUtil.setObject(ps, sqlParamList);
             rs = ps.executeQuery();
             while (rs.next()) {
@@ -122,7 +125,7 @@ public class GlobalSessionDBServiceImpl implements GlobalSessionService {
         } catch (SQLException e) {
             throw new StoreException(e);
         } finally {
-            IOUtil.close(rs, countRs);
+            IOUtil.close(rs, countRs, ps, countPs, conn);
         }
         return PageResult.success(list, count, param.getPageNum(), param.getPageSize());
     }

--- a/server/src/test/java/io/seata/server/lock/db/DataBaseLockManagerImplTest.java
+++ b/server/src/test/java/io/seata/server/lock/db/DataBaseLockManagerImplTest.java
@@ -68,9 +68,10 @@ public class DataBaseLockManagerImplTest {
 
     private static void prepareTable(BasicDataSource dataSource) {
         Connection conn = null;
+        Statement s = null;
         try {
             conn = dataSource.getConnection();
-            Statement s = conn.createStatement();
+            s = conn.createStatement();
             try {
                 s.execute("drop table lock_table");
             } catch (Exception e) {
@@ -81,7 +82,7 @@ public class DataBaseLockManagerImplTest {
         } catch (Exception e) {
             e.printStackTrace();
         } finally {
-            IOUtil.close(conn);
+            IOUtil.close(s, conn);
         }
     }
 
@@ -157,13 +158,14 @@ public class DataBaseLockManagerImplTest {
         Assertions.assertFalse(lockManager.acquireLock(branchSession3));
 
         String delSql = "delete from lock_table where xid in( 'abc-123:65867978' , 'abc-123:65867978' , 'abc-123:5678789'  )"  ;
-        Connection conn =  null;
+        Connection conn = null;
+        Statement stmt = null;
         try {
             conn = dataSource.getConnection();
-
-            conn.createStatement().execute(delSql);
+            stmt = conn.createStatement();
+            stmt.execute(delSql);
         } finally {
-            IOUtil.close(conn);
+            IOUtil.close(stmt, conn);
         }
     }
 
@@ -253,12 +255,13 @@ public class DataBaseLockManagerImplTest {
 
         String delSql = "delete from lock_table where xid in( 'abc-123:56877898' , 'abc-123:56877898' , 'abc-123:4575614354'  )"  ;
         Connection conn =  null;
+        Statement stmt = null;
         try {
             conn = dataSource.getConnection();
-
-            conn.createStatement().execute(delSql);
+            stmt = conn.createStatement();
+            stmt.execute(delSql);
         } finally {
-            IOUtil.close(conn);
+            IOUtil.close(stmt, conn);
         }
     }
 

--- a/server/src/test/java/io/seata/server/lock/db/DataBaseLockStoreDAOTest.java
+++ b/server/src/test/java/io/seata/server/lock/db/DataBaseLockStoreDAOTest.java
@@ -61,20 +61,21 @@ public class DataBaseLockStoreDAOTest {
 
     private static void prepareTable(BasicDataSource dataSource) {
         Connection conn = null;
+        Statement stmt = null;
         try {
             conn = dataSource.getConnection();
-            Statement s = conn.createStatement();
+            stmt = conn.createStatement();
             try {
-                s.execute("drop table lock_table");
+                stmt.execute("drop table lock_table");
             } catch (Exception e) {
             }
-            s.execute("CREATE TABLE lock_table ( xid varchar(96) ,  transaction_id long , branch_id long, resource_id varchar(32) ,table_name varchar(32) ,pk varchar(32) ,  row_key  varchar(128) primary key not null , status  integer , gmt_create TIMESTAMP(6) ,gmt_modified TIMESTAMP(6) ) ");
+            stmt.execute("CREATE TABLE lock_table ( xid varchar(96) ,  transaction_id long , branch_id long, resource_id varchar(32) ,table_name varchar(32) ,pk varchar(32) ,  row_key  varchar(128) primary key not null , status  integer , gmt_create TIMESTAMP(6) ,gmt_modified TIMESTAMP(6) ) ");
             System.out.println("create table lock_table success.");
 
         } catch (Exception e) {
             e.printStackTrace();
         } finally {
-            IOUtil.close(conn);
+            IOUtil.close(stmt, conn);
         }
     }
 
@@ -98,16 +99,17 @@ public class DataBaseLockStoreDAOTest {
 
         String sql = "select * from lock_table where xid = 'abc-123:123' and table_name  = 't' and row_key in ('abc-0','abc-1','abc-2')"  ;
         Connection conn =  null;
+        ResultSet rs = null;
         try{
             conn = dataSource.getConnection();
-            ResultSet rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            rs = conn.createStatement().executeQuery(sql);
+            if (rs.next()) {
                 Assertions.assertTrue(true);
-            }else {
+            } else {
                 Assertions.fail();
             }
         } finally {
-            IOUtil.close(conn);
+            IOUtil.close(rs, conn);
         }
 
         Assertions.assertTrue(dataBaseLockStoreDAO.unLock(lockDOs));
@@ -135,16 +137,17 @@ public class DataBaseLockStoreDAOTest {
 
         String sql = "select * from lock_table where xid = 'abc-123:123' and table_name  = 't' and row_key in ('abc-0','abc-1','abc-2')"  ;
         Connection conn =  null;
+        ResultSet rs = null;
         try{
             conn = dataSource.getConnection();
-            ResultSet rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            rs = conn.createStatement().executeQuery(sql);
+            if (rs.next()) {
                 Assertions.assertTrue(true);
-            }else {
+            } else {
                 Assertions.fail();
             }
         } finally {
-            IOUtil.close(conn);
+            IOUtil.close(rs, conn);
         }
 
         //lock again
@@ -174,12 +177,13 @@ public class DataBaseLockStoreDAOTest {
 
         String sql = "select * from lock_table where xid = 'abc-456:123' and table_name  = 't' and row_key in ('abc-0','abc-1','abc-2')"  ;
         Connection conn =  null;
+        ResultSet rs = null;
         try{
             conn = dataSource.getConnection();
-            ResultSet rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            rs = conn.createStatement().executeQuery(sql);
+            if (rs.next()) {
                 Assertions.assertTrue(true);
-            }else {
+            } else {
                 Assertions.fail();
             }
             rs.close();
@@ -188,14 +192,13 @@ public class DataBaseLockStoreDAOTest {
             Assertions.assertTrue(dataBaseLockStoreDAO.unLock(lockDOs));
 
             rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            if (rs.next()) {
                 Assertions.fail();
-            }else {
+            } else {
                 Assertions.assertTrue(true);
             }
-
         } finally {
-            IOUtil.close(conn);
+            IOUtil.close(rs, conn);
         }
 
 
@@ -244,16 +247,17 @@ public class DataBaseLockStoreDAOTest {
 
         String sql = "select * from lock_table where xid = 'abc-123:222' and table_name  = 't' and row_key in ('abc-0','abc-1','abc-2')"  ;
         Connection conn =  null;
+        ResultSet rs = null;
         try{
             conn = dataSource.getConnection();
-            ResultSet rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            rs = conn.createStatement().executeQuery(sql);
+            if (rs.next()) {
                 Assertions.assertTrue(true);
-            }else {
+            } else {
                 Assertions.fail();
             }
         } finally {
-            IOUtil.close(conn);
+            IOUtil.close(rs, conn);
         }
 
         List<LockDO> lockDOs_2 = new ArrayList<>();

--- a/server/src/test/java/io/seata/server/session/db/DataBaseSessionManagerTest.java
+++ b/server/src/test/java/io/seata/server/session/db/DataBaseSessionManagerTest.java
@@ -89,9 +89,10 @@ public class DataBaseSessionManagerTest {
 
     private static void prepareTable(BasicDataSource dataSource) {
         Connection conn = null;
+        Statement s = null;
         try {
             conn = dataSource.getConnection();
-            Statement s = conn.createStatement();
+            s = conn.createStatement();
             try {
                 s.execute("drop table global_table");
             } catch (Exception e) {
@@ -109,7 +110,7 @@ public class DataBaseSessionManagerTest {
         } catch (Exception e) {
             e.printStackTrace();
         } finally {
-            IOUtil.close(conn);
+            IOUtil.close(s, conn);
         }
     }
 

--- a/server/src/test/java/io/seata/server/store/db/LogStoreDataBaseDAOTest.java
+++ b/server/src/test/java/io/seata/server/store/db/LogStoreDataBaseDAOTest.java
@@ -64,9 +64,10 @@ public class LogStoreDataBaseDAOTest {
 
     private static void prepareTable(BasicDataSource dataSource) {
         Connection conn = null;
+        Statement s = null;
         try {
             conn = dataSource.getConnection();
-            Statement s = conn.createStatement();
+            s = conn.createStatement();
             try {
                 s.execute("drop table global_table");
             } catch (Exception e) {
@@ -86,7 +87,7 @@ public class LogStoreDataBaseDAOTest {
         } catch (Exception e) {
             e.printStackTrace();
         } finally {
-            IOUtil.close(conn);
+            IOUtil.close(s, conn);
         }
     }
 
@@ -123,12 +124,14 @@ public class LogStoreDataBaseDAOTest {
 
         String delSql = "delete from global_table where xid= 'abc-123:978786'";
         Connection conn = null;
-        try{
+        Statement stmt = null;
+        try {
             conn = dataSource.getConnection();
             //delete
-            conn.createStatement().execute(delSql);
-        }finally {
-            IOUtil.close(conn);
+            stmt = conn.createStatement();
+            stmt.execute(delSql);
+        } finally {
+            IOUtil.close(stmt, conn);
         }
 
     }
@@ -165,12 +168,14 @@ public class LogStoreDataBaseDAOTest {
 
         String delSql = "delete from global_table where xid= 'abc-123:978786'";
         Connection conn = null;
-        try{
+        Statement stmt = null;
+        try {
             conn = dataSource.getConnection();
             //delete
-            conn.createStatement().execute(delSql);
-        }finally {
-            IOUtil.close(conn);
+            stmt = conn.createStatement();
+            stmt.execute(delSql);
+        } finally {
+            IOUtil.close(stmt, conn);
         }
 
     }
@@ -236,11 +241,13 @@ public class LogStoreDataBaseDAOTest {
 
         String delSql = "delete from global_table where xid in ('abc-123:1267', 'abc-123:6978', 'abc-123:5657')";
         Connection conn = null;
-        try{
+        Statement stmt = null;
+        try {
             conn = dataSource.getConnection();
-            conn.createStatement().execute(delSql);
-        }finally {
-            IOUtil.close(conn);
+            stmt = conn.createStatement();
+            stmt.execute(delSql);
+        } finally {
+            IOUtil.close(stmt, conn);
         }
     }
 
@@ -303,11 +310,13 @@ public class LogStoreDataBaseDAOTest {
 
         String delSql = "delete from global_table where xid in ('abc-123:1267', 'abc-123:6978', 'abc-123:5657')";
         Connection conn = null;
-        try{
+        Statement stmt = null;
+        try {
             conn = dataSource.getConnection();
-            conn.createStatement().execute(delSql);
-        }finally {
-            IOUtil.close(conn);
+            stmt = conn.createStatement();
+            stmt.execute(delSql);
+        } finally {
+            IOUtil.close(stmt, conn);
         }
 
     }
@@ -331,19 +340,22 @@ public class LogStoreDataBaseDAOTest {
         String sql = "select * from global_table where xid= 'abc-123:333'";
         String delSql = "delete from global_table where xid= 'abc-123:333'";
         Connection conn = null;
-        try{
+        Statement stmt = null;
+        ResultSet rs = null;
+        try {
             conn = dataSource.getConnection();
-            ResultSet rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            stmt = conn.createStatement();
+            rs = stmt.executeQuery(sql);
+            if (rs.next()) {
                 Assertions.assertTrue(true);
-            }else{
+            } else{
                 Assertions.fail();
             }
 
-            conn.createStatement().execute(delSql);
+            stmt.execute(delSql);
 
-        }finally {
-            IOUtil.close(conn);
+        } finally {
+            IOUtil.close(rs, stmt, conn);
         }
     }
 
@@ -366,9 +378,11 @@ public class LogStoreDataBaseDAOTest {
         String sql = "select * from global_table where xid= 'abc-123:222'";
         String delSql = "delete from global_table where xid= 'abc-123:222'";
         Connection conn = null;
-        try{
+        Statement stmt = null;
+        try {
             conn = dataSource.getConnection();
-            ResultSet rs = conn.createStatement().executeQuery(sql);
+            stmt = conn.createStatement();
+            ResultSet rs = stmt.executeQuery(sql);
             if(rs.next()){
                 Assertions.assertTrue(true);
                 Assertions.assertEquals(1, rs.getInt("status"));
@@ -381,20 +395,20 @@ public class LogStoreDataBaseDAOTest {
             globalTransactionDO.setStatus(2);
             Assertions.assertTrue(logStoreDataBaseDAO.updateGlobalTransactionDO(globalTransactionDO));
 
-            rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            rs = stmt.executeQuery(sql);
+            if (rs.next()) {
                 Assertions.assertTrue(true);
                 Assertions.assertEquals(2, rs.getInt("status"));
-            }else{
+            } else{
                 Assertions.fail();
             }
             rs.close();
 
             //delete
-            conn.createStatement().execute(delSql);
+            stmt.execute(delSql);
 
-        }finally {
-            IOUtil.close(conn);
+        } finally {
+            IOUtil.close(stmt, conn);
         }
 
     }
@@ -418,13 +432,15 @@ public class LogStoreDataBaseDAOTest {
         String sql = "select * from global_table where xid= 'abc-123:222'";
         String delSql = "delete from global_table where xid= 'abc-123:222'";
         Connection conn = null;
-        try{
+        Statement stmt = null;
+        try {
             conn = dataSource.getConnection();
-            ResultSet rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            stmt = conn.createStatement();
+            ResultSet rs = stmt.executeQuery(sql);
+            if (rs.next()) {
                 Assertions.assertTrue(true);
                 Assertions.assertEquals(1, rs.getInt("status"));
-            }else{
+            } else {
                 Assertions.assertTrue(false);
             }
             rs.close();
@@ -433,20 +449,20 @@ public class LogStoreDataBaseDAOTest {
             globalTransactionDO.setStatus(2);
             Assertions.assertTrue(logStoreDataBaseDAO.updateGlobalTransactionDO(globalTransactionDO,1));
 
-            rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            rs = stmt.executeQuery(sql);
+            if (rs.next()) {
                 Assertions.assertTrue(true);
                 Assertions.assertEquals(2, rs.getInt("status"));
-            }else{
+            } else{
                 Assertions.assertTrue(false);
             }
             rs.close();
 
             //delete
-            conn.createStatement().execute(delSql);
+            stmt.execute(delSql);
 
-        }finally {
-            IOUtil.close(conn);
+        } finally {
+            IOUtil.close(stmt, conn);
         }
 
     }
@@ -474,17 +490,19 @@ public class LogStoreDataBaseDAOTest {
 
         String sql = "select * from global_table where xid= 'abc-123:555'";
         Connection conn = null;
-        try{
+        Statement stmt = null;
+        try {
             conn = dataSource.getConnection();
-            ResultSet rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            stmt = conn.createStatement();
+            ResultSet rs = stmt.executeQuery(sql);
+            if (rs.next()) {
                 Assertions.fail();
-            }else{
+            } else{
                 Assertions.assertTrue(true);
             }
             rs.close();
-        }finally {
-            IOUtil.close(conn);
+        } finally {
+            IOUtil.close(stmt, conn);
         }
     }
 
@@ -531,23 +549,25 @@ public class LogStoreDataBaseDAOTest {
         Assertions.assertTrue(CollectionUtils.isNotEmpty(rets));
         Assertions.assertEquals(2, rets.size());
 
-        if(78563453 == rets.get(0).getBranchId() && 345465676 == rets.get(1).getBranchId()){
+        if (78563453 == rets.get(0).getBranchId() && 345465676 == rets.get(1).getBranchId()) {
             Assertions.assertTrue(true);
-        }else if(78563453 == rets.get(1).getBranchId() && 345465676 == rets.get(0).getBranchId()){
+        } else if (78563453 == rets.get(1).getBranchId() && 345465676 == rets.get(0).getBranchId()) {
             Assertions.assertTrue(true);
-        }else {
+        } else {
             Assertions.fail();
         }
 
         String delSql = "delete from branch_table where xid= 'abc-123:6789' ";
         Connection conn = null;
-        try{
+        Statement stmt = null;
+        try {
             conn = dataSource.getConnection();
 
-            conn.createStatement().execute(delSql);
+            stmt = conn.createStatement();
+            stmt.execute(delSql);
 
-        }finally {
-            IOUtil.close(conn);
+        } finally {
+            IOUtil.close(stmt, conn);
         }
     }
 
@@ -573,19 +593,21 @@ public class LogStoreDataBaseDAOTest {
         String sql = "select * from branch_table where xid= 'abc-123:7777' and branch_id = 1234508";
         String delSql = "delete from branch_table where xid= 'abc-123:7777' and branch_id = 1234508";
         Connection conn = null;
-        try{
+        Statement stmt = null;
+        try {
             conn = dataSource.getConnection();
 
-            ResultSet rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            stmt = conn.createStatement();
+            ResultSet rs = stmt.executeQuery(sql);
+            if (rs.next()) {
                 Assertions.assertTrue(true);
-            }else {
+            } else {
                 Assertions.fail();
             }
 
-            conn.createStatement().execute(delSql);
-        }finally {
-            IOUtil.close(conn);
+            stmt.execute(delSql);
+        } finally {
+            IOUtil.close(stmt, conn);
         }
 
     }
@@ -614,20 +636,22 @@ public class LogStoreDataBaseDAOTest {
         String sql = "select * from branch_table where xid= 'abc-123:8888' and branch_id = 343434318";
         String delSql = "delete from branch_table where xid= 'abc-123:8888' and branch_id = 343434318";
         Connection conn = null;
-        try{
+        Statement stmt = null;
+        try {
             conn = dataSource.getConnection();
 
-            ResultSet rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            stmt = conn.createStatement();
+            ResultSet rs = stmt.executeQuery(sql);
+            if (rs.next()) {
                 Assertions.assertTrue(true);
                 Assertions.assertEquals(3, rs.getInt("status"));
-            }else {
+            } else {
                 Assertions.fail();
             }
 
-            conn.createStatement().execute(delSql);
-        }finally {
-            IOUtil.close(conn);
+            stmt.execute(delSql);
+        } finally {
+            IOUtil.close(stmt, conn);
         }
     }
 
@@ -653,13 +677,15 @@ public class LogStoreDataBaseDAOTest {
         String sql = "select * from branch_table where xid= 'abc-123:9999' and branch_id = 34567798";
         String delSql = "delete from branch_table where xid= 'abc-123:9999' and branch_id = 34567798";
         Connection conn = null;
-        try{
+        Statement stmt = null;
+        try {
             conn = dataSource.getConnection();
 
-            ResultSet rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            stmt = conn.createStatement();
+            ResultSet rs = stmt.executeQuery(sql);
+            if (rs.next()) {
                 Assertions.assertTrue(true);
-            }else {
+            } else {
                 Assertions.fail();
             }
             rs.close();
@@ -667,16 +693,16 @@ public class LogStoreDataBaseDAOTest {
             //delete
             logStoreDataBaseDAO.deleteBranchTransactionDO(branchTransactionDO);
 
-            rs = conn.createStatement().executeQuery(sql);
-            if(rs.next()){
+            rs = stmt.executeQuery(sql);
+            if (rs.next()) {
                 Assertions.fail();
-            }else {
+            } else {
                 Assertions.assertTrue(true);
             }
             rs.close();
 
-        }finally {
-            IOUtil.close(conn);
+        } finally {
+            IOUtil.close(stmt, conn);
         }
     }
 

